alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/gui/effects/DrawCognitiveMap.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary.gui.effects;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import it.unibo.alchemist.boundary.wormhole.interfaces.IWormhole2D;
+import it.unibo.alchemist.model.implementations.geometry.euclidean.twod.Ellipse;
+import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition;
+
+import it.unibo.alchemist.model.interfaces.OrientingPedestrian;
+import it.unibo.alchemist.model.interfaces.Environment;
+import it.unibo.alchemist.model.interfaces.Environment2DWithObstacles;
+import it.unibo.alchemist.model.interfaces.Node;
+import it.unibo.alchemist.model.interfaces.Position2D;
+import it.unibo.alchemist.model.interfaces.graph.NavigationGraph;
+import org.danilopianini.lang.RangedInteger;
+import org.danilopianini.view.ExportForGUI;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.Shape;
+import java.awt.geom.Ellipse2D;
+import java.awt.geom.Point2D;
+import java.awt.geom.Rectangle2D;
+import java.util.Optional;
+
+/**
+ * Draws an orienting pedestrian' cognitive map.
+ */
+public class DrawCognitiveMap implements Effect {
+
+    /**
+     */
+    protected static final int MAX_COLOUR_VALUE = 255;
+    /**
+     */
+    protected static final int INITIAL_ALPHA_DIVIDER = 5;
+    /**
+     */
+    protected static final Logger L = LoggerFactory.getLogger(DrawShape.class);
+    private static final long serialVersionUID = 1L;
+    @ExportForGUI(nameToExport = "A")
+    private RangedInteger alpha = new RangedInteger(0, MAX_COLOUR_VALUE, MAX_COLOUR_VALUE / INITIAL_ALPHA_DIVIDER);
+    @ExportForGUI(nameToExport = "R")
+    private RangedInteger red = new RangedInteger(0, MAX_COLOUR_VALUE, MAX_COLOUR_VALUE);
+    @ExportForGUI(nameToExport = "G")
+    private RangedInteger green = new RangedInteger(0, MAX_COLOUR_VALUE);
+    @ExportForGUI(nameToExport = "B")
+    private RangedInteger blue = new RangedInteger(0, MAX_COLOUR_VALUE);
+    private Color colorCache = Color.RED;
+    @SuppressFBWarnings("SE_TRANSIENT_FIELD_NOT_RESTORED")
+    private transient NavigationGraph<? extends Euclidean2DPosition, ?, Ellipse, ?> cognitiveMap;
+    @SuppressFBWarnings("SE_TRANSIENT_FIELD_NOT_RESTORED")
+    private transient Optional<Node> markerNode = Optional.empty();
```

Comment:
This is a bug! Either mark it @nullable and remove the optional, or use Guava's Optional (they are serializable)

Reply:
Better: use non-transient int and get the node via enviroment